utils/merge_test.go

```diff
@@ -0,0 +1,581 @@
+package utils
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMergeWithStructs(t *testing.T) {
```

Comment:
One thing I don't think I'm seeing by way of coverage is a guarantee that pointers inside structs aren't being copied over nor that pointers inside the base or patch structs coming in aren't being modified in some way. Is that easy to add and thus verify?